lib/api/src/rt/v8/entities/exception.rs

```diff
@@ -0,0 +1,63 @@
+//! Data types, functions and traits for `sys` runtime's `Tag` implementation.
+use std::any::Any;
+
+use wasmer_types::{TagType, Type};
+
+use crate::{
+    v8::vm::{VMException, VMExceptionRef},
+    AsStoreMut, AsStoreRef, Tag, Value,
+};
+
+use super::store::StoreHandle;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+/// A WebAssembly `tag` in the `v8` runtime.
+pub struct Exception {
+    pub(crate) handle: VMException,
+}
+
+unsafe impl Send for Exception {}
+unsafe impl Sync for Exception {}
+
+impl Exception {
+    /// Create a new [`Exception`].
+    pub fn new(store: &mut impl AsStoreMut, tag: Tag, payload: &[Value]) -> Self {
+        todo!()
+    }
+}
+
+#[derive(Debug, Clone)]
+#[repr(transparent)]
+/// A WebAssembly `extern ref` in `v8`.
+pub struct ExceptionRef;
```

Comment:
This should be opaque (pub only for the crate)

Reply:
[6b716f5](https://github.com/wasmerio/wasmer/pull/5347/commits/6b716f57769ad520116ec0f997a8d8bee79978e9)